corva/loader/base.py

```diff
@@ -1,20 +1,18 @@
-import json
-from abc import abstractmethod, ABC
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar
 
-from corva.types import EVENT_TYPE
-from corva.event import Event
+from pydantic import parse_raw_as
+
+from corva.models.base import BaseEvent
 
 
 class BaseLoader(ABC):
+    parse_as_type: ClassVar[Any]
+
     @abstractmethod
-    def load(self, event: str) -> Event:
+    def load(self, event: str) -> BaseEvent:
         pass
 
-    @staticmethod
-    def _load_json(event: str) -> EVENT_TYPE:
-        try:
-            event = json.loads(event)
-        except ValueError as exc:
-            raise ValueError('Invalid event JSON') from exc
-
-        return event
+    @classmethod
+    def parse(cls, event: str) -> Any:
```

Comment:
I doubt we want to return `Any` here. Maybe a union of multiple event types instead?

Reply:
Also, I am not sure that this belongs as an inherited class. Can we have the model classes themselves each just be able to parse their own event?

Reply:
moved loading logic to event classes